src/Microsoft.AspNetCore.Razor.Language/DefaultRazorProjectEngine.cs

```diff
@@ -138,12 +138,20 @@ private void ConfigureDesignTimeCodeGenerationOptions(RazorCodeGenerationOptions
         }
 
         // Internal for testing
-        internal static RazorSourceDocument ConvertToSourceDocument(RazorProjectItem importItem)
+        internal static RazorSourceDocument ConvertToSourceDocument(RazorProjectItem importItem, bool designTime = false)
         {
             if (importItem.Exists)
             {
-                // Normal import, has file paths, content etc.
-                return RazorSourceDocument.ReadFrom(importItem);
+                try
+                {
+                    // Normal import, has file paths, content etc.
+                    return RazorSourceDocument.ReadFrom(importItem);
+                }
+                catch (IOException) when (designTime)
+                {
+                    // Something happened when trying to read the item from disk.
+                    // Catch the exception so we don't crash the editor.
```

Comment:
Should log another issue to report this as an error to the user, otherwise this is a silent error that the user will have a verrrry hard time diagnosing. Given how we open files reporting an error could suffice for design time and runtime; reason being we'll typically only fail to open a file if it's permissions are whack (something a user should fix).

Reply:
Yep. We'd need to add UI for that, so it goes to 15.8. We'll do it when it's a priority

Reply:
I'm going to file a broader issue which is basically "Add a way to flow Razor editor parsing errors to the IDE" because we don't have that.

Reply:
Filed https://github.com/aspnet/Razor/issues/2193